src/api/api_missions.nit

```diff
@@ -35,6 +36,40 @@ end
 
 class APIMission
 	super MissionHandler
+	super AuthHandler
+
+	redef fun post(req, res) do
+		var player = get_player(req, res)
+		if player == null then return
+		var mission = get_mission(req, res)
+		if mission == null then return
+		var post = req.post_args
+		if not post.has_key("source") then
+			res.error 400
+			print "No code provided for submission"
+			return
+		end
+		var code = post["source"]
+		if not post.has_key("lang") then
+			res.error 400
+			print "No lang provided for submission"
```

Comment:
Should the mission (or the track) be used to deduce the lang? Why the submitter could choose to send C++ to pep8 mission?